File: Common/AppGroup.m
//
//  AppGroup.m
//  FlickSKK
//
//  Created by BAN Jun on 2014/10/26.
//  Copyright (c) 2014年 BAN Jun. All rights reserved.
//

#import "AppGroup.h"

#define STR(x) @#x
#define STR2(x) STR(x)
static NSString * const kAppIdentifier = STR2(APP_IDENTIFIER);
static NSString * const kInitalText = STR2(INITIAL_TEXT);

@implementation AppGroup

+ (NSString *)appGroupID
{
    return [NSString stringWithFormat:@"group.%@", kAppIdentifier];
}

+ (NSString *)initialText
{
    return kInitalText;
}

+ (NSString *)pathForResource:(NSString *)subpath
{
    return [self urlForResource:subpath].path;
}

+ (NSURL *)urlForResource:(NSString *)subpath {
    NSURL *container = [[NSFileManager defaultManager] containerURLForSecurityApplicationGroupIdentifier:[self appGroupID]];
    return [container URLByAppendingPathComponent:subpath];
}

@end

File: FlickSKK/AppGroup.swift
//
//  AppGroup.swift
//  FlickSKK
//
//  Created by BAN Jun on 2014/10/19.
//  Copyright (c) 2014年 BAN Jun. All rights reserved.
//

import Foundation


class AppGroup {
    class func pathForResource(subpath: String) -> String? {
        let userName = AppGroupSupport.userName()
        let identifier = "group.org.codefirst.skk.\(userName).FlickSKK"
        return NSFileManager.defaultManager().containerURLForSecurityApplicationGroupIdentifier(identifier)?.path?.stringByAppendingPathComponent(subpath)
    }
}